lib/base/json.cpp

```diff
@@ -166,16 +166,18 @@ void Encode(JsonEncoder<prettyPrint>& stateMachine, const Value& value)
 					}
 				}
 
-				Array::Ptr arr = dynamic_pointer_cast<Array>(obj);
-				if (arr) {
-					EncodeArray(stateMachine, arr);
-					break;
+				{
+					Array::Ptr arr = dynamic_pointer_cast<Array>(obj);
+					if (arr) {
+						EncodeArray(stateMachine, arr);
+						break;
+					}
 				}
-			}
 
-			stateMachine.Null();
-
-			break;
+				// obj is most likely a function => "Object of type 'Function'"
```

Comment:
This is not about likeliness but about what you put in. I'd just say "other object types are encoded as their string representation, for example "Object of type 'Function'"

Reply:
According to Eric it **is**.